File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        left = right = minJumps = 0
        
        while right < len(nums) - 1:
            furthest = 0
            
            for i in range(left, right + 1):
                furthest = max(furthest, i + nums[i])
            left = right + 1
            right = furthest
            minJumps += 1
        return minJumps
        
        

File: 0045-jump-game-ii/0045-jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        int left = 0;
        int right = 0;
        int minJumps = 0;
        
        while(right < nums.length - 1){
            int furthest = 0;
            
            for(int i = left; i < right + 1; i++){
                furthest = Math.max(furthest, i + nums[i]);
                
            }
            left = right + 1;
            right = furthest;
            minJumps++;
        }
        return minJumps;
    }
}